File: src/Models/NDVI/NDVI_forcing.jl
export NDVI_forcing

struct NDVI_forcing <: NDVI end

function compute(params::NDVI_forcing, forcing, land, helpers)
    ## unpack forcing
    @unpack_nt f_NDVI ⇐ forcing

    NDVI = f_NDVI

    ## pack land variables
    @pack_nt NDVI ⇒ land.states
    return land
end

purpose(::Type{NDVI_forcing}) = "sets the value of land.states.NDVI from the forcing in every time step"

@doc """

$(getBaseDocString(NDVI_forcing))

---

# Extended help

*References*

*Versions*
 - 1.0 on 29.04.2020 [sbesnard]

*Created by:*
 - sbesnard
"""
NDVI_forcing

File: model/modules/NDVI/NDVI_forcing/NDVI_forcing.m
function [f,fe,fx,s,d,p]=NDVI_forcing(f,fe,fx,s,d,p,info,tix)
% +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
% sets the value of s.cd.NDVI from the forcing in every time step
%
% Inputs:
%   - f.NDVI read from the forcing data set
%
% Outputs:
%   - s.cd.NDVI: the value of NDVI for current time step
%
% Modifies:
%   - s.cd.NDVI
%
% References:
%   - 
%
% Created by:
%   - Simon Besnard (sbesnard)
%
% Versions:
%   - 1.0 on 29.04.2020 (sbesnard): 
%
% +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

%% 
s.cd.NDVI = f.NDVI(:,tix);

end